src/Data/Trees/KdTree.hs

```diff
@@ -154,7 +154,7 @@ remove :: (Eq p, Point p) => KdTree p -> p -> KdTree p
 remove KdEmpty _ = KdEmpty
 remove (KdNode l p r axis) pKill
   | p == pKill = fromListWithDepth (toList l ++ toList r) axis
-  | coord axis pKill <= coord axis p = KdNode (remove l pKill) p r axis
```

Comment:
I didn't myself write this code, thus this question: Shouldn't `coord axis pKill == coord axis p` imply `p == pKill`? Or does it actually mean the point is in the right half-space?

Reply:
coord axis pKill == coord axis p doesn't imply p == pKill. p == pKill means 
(coord 0 p == coord 0 pKill) && (coord 1 p == coord 1 pKill ) && (coord 2 p == coord 2 pKill) .

76 Line:
leftPoints = filter (\p -> coord axis p < medianCoordinate) sortedPoints

so, equal node should be right.